spec/tbd_tests_spec.rb

```diff
@@ -11543,4 +11549,570 @@
+      loops     = TBD.airLoopsHVAC?(model)
+      setpoints = TBD.heatingTemperatureSetpoints?(model)
+      setpoints = TBD.coolingTemperatureSetpoints?(model) || setpoints
+      expect(model.getConstructions.empty?).to be(false)
+
+      expect(setpoints).to be( true)
+      expect(loops    ).to be(false)
+
+      # No point here in replacing attic space type with "plenum" - a last resort
+      # check, ignored here by TBD as there are valid setpoints set elsewhere in
+      # the model. Instead, temporarily add a heating dual setpoint schedule to
+      # the attic zone thermostat (yet without valid schedule temperatures).
+      attic = model.getSpaceByName("attic")
+      expect(attic.empty?              ).to be(false)
+      attic = attic.get
+      expect(attic.partofTotalFloorArea).to be(false)
+      expect(attic.thermalZone.empty?  ).to be(false)
+      zone  = attic.thermalZone.get
+      expect(zone.isPlenum             ).to be(false)
+      tstat = zone.thermostat
+      expect(tstat.empty?              ).to be(false)
+      tstat = tstat.get
+
+      expect(tstat.to_ThermostatSetpointDualSetpoint.empty?).to be(false)
+      tstat = tstat.to_ThermostatSetpointDualSetpoint.get
+
+      # Before the addition.
+      expect(tstat.getHeatingSchedule.empty?).to be(true)
+      expect(tstat.getCoolingSchedule.empty?).to be(true)
+
+      heat = TBD.maxHeatScheduledSetpoint(zone)
+      cool = TBD.minCoolScheduledSetpoint(zone)
+
+      expect(heat.nil?       ).to be(false)
+      expect(cool.nil?       ).to be(false)
+      expect(heat.is_a?(Hash)).to be( true)
+      expect(cool.is_a?(Hash)).to be( true)
+      expect(heat.key?(:spt )).to be( true)
+      expect(cool.key?(:spt )).to be( true)
+      expect(heat.key?(:dual)).to be( true)
+      expect(cool.key?(:dual)).to be( true)
+      expect(heat[:spt ].nil?).to be( true)
+      expect(cool[:spt ].nil?).to be( true)
+      expect(heat[:dual]     ).to be(false)
+      expect(cool[:dual]     ).to be(false)
+
+      expect(TBD.plenum?(attic, loops, setpoints)).to be(false)
+
+      # Add a dual setpoint temperature schedule.
+      identifier = "TEMPORARY attic setpoint schedule"
+      sched = OpenStudio::Model::ScheduleCompact.new(model)
+      sched.setName(identifier)
+      expect(sched.constantValue.empty?                        ).to be(true)
+      expect(tstat.setHeatingSetpointTemperatureSchedule(sched)).to be(true)
+
+      # After the addition.
+      expect(tstat.getHeatingSchedule.empty?).to be(false)
+      expect(tstat.getCoolingSchedule.empty?).to be( true)
+      heat = TBD.maxHeatScheduledSetpoint(zone)
+
+      expect(heat.nil?       ).to be(false)
+      expect(heat.is_a?(Hash)).to be( true)
+      expect(heat.key?(:spt )).to be( true)
+      expect(heat.key?(:dual)).to be( true)
+      expect(heat[:spt ].nil?).to be( true)
+      expect(heat[:dual]     ).to be( true)
+
+      expect(TBD.plenum?(attic, loops, setpoints)).to be(true) # works ...
```

Comment:
With heating/cooling setpoints set (elsewhere in the model), the straightforward "plenum" spacetype check/switch is ignored here. Temporarily adding an (incomplete) heating setpoint schedule is enough for TBD to now identify the space as an indirectly conditioned plenum.